src/main/java/teammates/ui/webapi/GetDeadlineExtensionAction.java

```diff
@@ -29,19 +33,40 @@ public JsonResult execute() {
         String userEmail = getNonNullRequestParamValue(Const.ParamsNames.USER_EMAIL);
         boolean isInstructor = Boolean.parseBoolean(getNonNullRequestParamValue(Const.ParamsNames.IS_INSTRUCTOR));
 
-        DeadlineExtensionAttributes deadlineExtension =
-                logic.getDeadlineExtension(courseId, feedbackSessionName, userEmail, isInstructor);
+        if (isCourseMigrated(courseId)) {
+            FeedbackSession feedbackSession = getNonNullSqlFeedbackSession(feedbackSessionName, courseId);
+            User user = isInstructor
+                    ? sqlLogic.getInstructorForEmail(courseId, userEmail)
+                    : sqlLogic.getStudentForEmail(courseId, userEmail);
 
-        if (deadlineExtension == null) {
-            throw new EntityNotFoundException(
-                    "Deadline extension for course id: " + courseId
-                    + " and feedback session name: " + feedbackSessionName
-                    + " and " + (isInstructor ? "instructor" : "student")
-                    + " email: " + userEmail
-                    + " not found.");
-        }
+            Instant deadlineExtensionEndTime = sqlLogic.getExtendedDeadlineForUser(feedbackSession, user);
 
-        return new JsonResult(new DeadlineExtensionData(deadlineExtension));
-    }
+            if (deadlineExtensionEndTime == null) {
+                throw new EntityNotFoundException(
+                        "Deadline extension for course id: " + courseId
+                        + " and feedback session name: " + feedbackSessionName
+                        + " and " + (isInstructor ? "instructor" : "student")
+                        + " email: " + userEmail
+                        + " not found.");
+            }
+
+            // set sentClosingEmail as false by default since it is removed.
+            return new JsonResult(new DeadlineExtensionData(courseId, feedbackSessionName,
```

Comment:
@samuelfangjw @hhdqirui Im not sure why sentClosingEmail was removed from FeedbackSession

Reply:
see https://github.com/TEAMMATES/teammates/issues/11728